apiserver/facades/client/bundle/bundle.go

```diff
@@ -11,43 +11,105 @@ import (
 	"github.com/juju/errors"
 	"github.com/juju/loggo"
 	"gopkg.in/juju/charm.v6"
+	"gopkg.in/juju/names.v2"
 
 	"github.com/juju/juju/apiserver/common"
 	"github.com/juju/juju/apiserver/facade"
 	"github.com/juju/juju/apiserver/params"
 	"github.com/juju/juju/constraints"
-	"github.com/juju/juju/state"
+	"github.com/juju/juju/permission"
 	"github.com/juju/juju/storage"
 )
 
+var logger = loggo.GetLogger("juju.apiserver.bundle")
+
+// APIv1 provides the Bundle API facade for version 1.
+type APIv1 struct {
+	*APIv2
+}
+
+// APIv2 provides the Bundle API facade for version 2.
+type APIv2 struct {
+	*BundleAPI
+}
+
+// BundleAPI implements the Bundle interface and is the concrete implementation
+// of the API end point.
+type BundleAPI struct {
+	backend    Backend
+	authorizer facade.Authorizer
+	modelTag   names.ModelTag
+}
+
+// NewFacadeV1 provides the signature required for facade registration
+// version 1.
+func NewFacadeV1(ctx facade.Context) (*APIv1, error) {
+	api, err := NewFacadeV2(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &APIv1{api}, nil
+}
+
+// NewFacadeV2 provides the signature required for facade registration
+// for version 2.
+func NewFacadeV2(ctx facade.Context) (*APIv2, error) {
+	api, err := newFacade(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &APIv2{api}, nil
+}
+
 // NewFacade provides the required signature for facade registration.
-func NewFacade(_ *state.State, _ facade.Resources, auth facade.Authorizer) (Bundle, error) {
-	return NewBundle(auth)
+func newFacade(ctx facade.Context) (*BundleAPI, error) {
+	authorizer := ctx.Auth()
+	st := ctx.State()
+	model, err := st.Model()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return NewBundleAPI(
+		st,
+		authorizer,
+		model.ModelTag(),
```

Comment:
If we are just after the model tag from the model, we don't need to load the model from state, but there is a model tag attribute on the state object.